### ISSUE TITLE
Can not get PVC automatically deleted on RSC chart uninstall

### ISSUE THREAD

**Issue by helgard88** [rstudio/helm#492]
Currently `sharedStorage.annotations` default value is `{"helm.sh/resource-policy":"keep"}`. That means that PVC should not be deleted on helm uninstall. 
We use automated pipeline with Terraform Kitchen for our testing environment that requires automated destroy of an infrastructure. But in case PVC still exists, PV can not be deleted, and the pipeline crashes on destroy stage.
I tried to pass both an empty value and  `{"helm.sh/resource-policy":"delete"}` into `sharedStorage.annotations` but it did not work. The PVC is still there after the RSC chart gets uninstalled. Is it an expected behaviour? Maybe there are some other workarounds to get the PVC automatically deleted?

**Comment by tnederlof** [comment]
@helgard88 when you set `sharedStorage.annotations`, did you remove your current deployment and delete the PVC before trying again? Does the annotation show up as expected when you check describe on the new PVC but the behavior is not what you expect or does the annotation not show up as changed at all?

**Comment by dbkegley** [comment]
I don't think this is related to our helm chart, sounds like this is a broader issue with helm resource cleanup

https://github.com/helm/helm/issues/5156#issuecomment-1822974212
